### PR TITLE
Calculate tail probabilities explicitly in anomaly_likelihood

### LINKED ISSUE #3170: Anomaly likelihood calculation of tail probability should use Q function, not lookup table
Using a lookup table lacks precision, and is smelly code considering we can easily calculate the probabilities (with built-in functions).

This is related to [NAB PR #251](https://github.com/numenta/NAB/pull/251).

### PR DESCRIPTION
**Pull request by BoltzmannBrain**
Fixes #3170 
Running the Numenta detectors in NAB shows negligible differences:

```
     "numenta": {
-        "reward_low_FN_rate": 69.37703104097989,
-        "reward_low_FP_rate": 58.58644756021552,
-        "standard": 65.27244311318965
+        "reward_low_FN_rate": 69.436217952925276,
+        "reward_low_FP_rate": 58.759846356594835,
+        "standard": 65.361223481120689
     },
     "numentaTM": {
-        "reward_low_FN_rate": 66.09146564859769,
-        "reward_low_FP_rate": 52.389446324344824,
-        "standard": 61.20616399012932
+        "reward_low_FN_rate": 65.90483945238218,
+        "reward_low_FP_rate": 52.366094841866385,
+        "standard": 60.926224695818966
     }
```

Replaces #3171

### CODE REVIEW

**Comment by rhyolight** [comment]
@BoltzmannBrain Wanna tag someone to review?

**Comment by BoltzmannBrain** [comment]
Sure. I've been using this PR to test the CI validation hook, but there are better ways to do this 😀 
@subutai please take a look.

**Comment by BoltzmannBrain** [comment]
@oxtopus please review?

File: src/nupic/algorithms/anomaly_likelihood.py [hunk 1]
```diff
@@ -620,23 +620,24 @@ def nullDistribution(verbosity=0):
...
 
-  xs = round(xs)
-  if xs > 70:
-    return 0.0
-  else:
-    return Q[xs]
+  # Calculate the Q function with the complementary error function, explained
+  # here: http://www.gaussianwaves.com/2012/07/q-function-and-error-functions
+  z = (x - distributionParams["mean"]) / distributionParams["stdev"]
+  return 0.5 * math.erfc(z/math.sqrt(2))
```
Comment (oxtopus) [inline src/nupic/algorithms/anomaly_likelihood.py]: Consider replacing `math.sqrt(2)` with the hard-coded value to avoid the function-call penalty.

**Comment by BoltzmannBrain** [comment]
@oxtopus good to go?

**Comment by oxtopus** [comment]
👍

**Pull request merged**